core/src/main/java/com/linecorp/armeria/internal/common/InitiateConnectionShutdown.java

```diff
@@ -0,0 +1,17 @@
+package com.linecorp.armeria.internal.common;
+
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+public final class InitiateConnectionShutdown {
+    @Nullable
+    private final Duration gracePeriod;
```

Comment:
How about storing this as a primitive value such as `long` so we don't have to create two objects for this event? We could use a negative value as 'use the default'.

Reply:
See https://github.com/line/armeria/pull/3715/files#r676900302

> We could use a negative value as 'use the default'.

There's a downside - users of the API won't be able to use duration arithmetics directly, e.g. I'd expect `InitiateConnectionShutdown(Duration.between(now, deadline))` to shutdown connection w/o grace period, not use default.

Reply:
Wanted to note that I also don't like using null. If that's your concern - I also considered using Optional<Duration> to make it explicit that Duration may not be provided, in which case we fallback to the default. If you think it's better than using @Nullable gracePeriod - let me know and I'll update to use Optional

Reply:
> users of the API won't be able to use duration arithmetics directly

My suggestion was to provide the API that accepts both `long` and `Duration` to our users, but use `long` for internal representation, so users can still use duration arithmetics.

Using `null` is not a concern; we actually prefer using `@Nullable` than using `Optional` due to performance reasons (increased GC pression when JIT fails to do escape analysis properly).

Reply:
Discussed offline, updated to store drain period as long internally to make sure it doesn't cause GC issues in cases where Armeria needs to close large number of connections.